llvm/lib/Target/SPIRV/SPIRVPreLegalizer.cpp

```diff
@@ -379,6 +379,89 @@ static void processInstrsWithTypeFolding(MachineFunction &MF,
         processInstr(MI, MIB, MRI, GR);
 }
 
+static void processSwitches(MachineFunction &MF, SPIRVGlobalRegistry *GR) {
+  DenseMap<Register, SmallDenseMap<uint64_t, MachineBasicBlock *>>
+      SwitchRegToMBB;
+  DenseMap<Register, MachineBasicBlock *> DefaultMBBs;
+  DenseSet<Register> SwitchRegs;
+  auto &MRI = MF.getRegInfo();
+  MachineIRBuilder MIB(MF);
+
+  // Walk MIs and collect information about destination MBBs to update
+  // spv_switch call. We assume that all spv_switch precede corresponding ICMPs.
```

Comment:
Which ICMPs is this about? IIUC at this point `llvm.switch`s aren't lowered to a set of icmp+br since this is still PreLegalizer

Reply:
This pass works after IRTranslator, so switches are lowered to icmp+br.